.github/workflows/github_job_scheduler.py

```diff
@@ -0,0 +1,212 @@
+                             "execute-matrix -> Remove executable code from non-relevant files in current offset. "
+                             "clean-html -> Delete html files built that are not relevant for current matrix offset.",
+                        choices=["build-matrix", "execute-matrix", "clean-html"],
+                        type=str)
+    parser.add_argument("directory",
+                        help="The path to the qml directory",
+                        type=str)
+    parser.add_argument("--examples-dir",
+                        help="The directory where sphinx documents exists. Similar to Sphinx examples-dir."
+                             "Default: 'demonstrations'",
+                        default="demonstrations")
+    parser.add_argument("--num-workers",
+                        help="The total number of jobs the build should be split across",
+                        type=int,
+                        required=True)
+    parser.add_argument("--glob-pattern",
+                        help="The pattern to use to search for files in the examples-dir."
+                             "Default: '*.py'",
+                        type=str,
+                        default="*.py")
+    parser_build = parser.add_argument_group("Build Matrix")
+
+    parser_execute = parser.add_argument_group("Execute Matrix / Clean")
+    parser_execute.add_argument("--offset",
+                                help="The current matrix output to retain files from",
+                                type=int)
+    parser_execute.add_argument("--dry-run",
+                                help="Will not delete files but list what will be deleted",
+                                action="store_true")
+    parser_execute.add_argument("--preserve-non-sphinx-images",
+                                help="Do not delete static images in the gallery _images directory",
+                                action="store_true")
+    parser_execute.add_argument("--verbose",
+                                help="Print name of files that are being cleaned up for current offset",
+                                action="store_true")
+
+    parser_results = parser.parse_args()
+
+    directory_qml = PosixPath(parser_results.directory)
+    directory_examples = directory_qml / parser_results.examples_dir
+    assert directory_examples.exists(), f"Could not find {parser_results.examples_dir!r} folder " \
+                                        f"under {parser_results.directory!r}"
+
+    worker_count = parser_results.num_workers
+    assert worker_count > 0, "Total number of workers has to be greater than 1"
+
+    if parser_results.action == "clean-html":
+        output = clean_html(worker_count, directory_qml, directory_examples, parser_results)
+    else:
+        output = {
+            "build-matrix": build_matrix,
+            "execute-matrix": execute_matrix
+        }[parser_results.action](worker_count, directory_examples, parser_results)
```

Comment:
Minor: just for improved readability.
```suggestion
        action_dict = {
            "build-matrix": build_matrix,
            "execute-matrix": execute_matrix
        }
        output = action_dict[parser_results.action](worker_count, directory_examples, parser_results)
```

Reply:
Updated